pubtools/pulplib/_impl/client/search.py

```diff
@@ -79,12 +90,15 @@ def field_match(to_match):
         return {"$regex": to_match._pattern}
 
     if isinstance(to_match, EqMatcher):
-        return {"$eq": to_match._value}
+        return {"$eq": to_mongo_json(to_match._value)}
 
     if isinstance(to_match, InMatcher):
-        return {"$in": to_match._values}
+        return {"$in": to_mongo_json(to_match._values)}
```

Comment:
I don't think this is correct, seems like it should rather be converting each value *in the list*.  For example, if you had a list containing:

```
['somestr', 123, datetime.datetime(...)]
```

Then I think you want the mongo conversion not to just return you the unmodified object but rather to return you:

```
['somestr', 123, {"$date": "..."}]
```

I suppose the correct place to handle this is to leave this code alone and do it inside of to_mongo_json, so I've made a suggestion there.